modules/fbs-core/api/src/main/kotlin/de/thm/ii/fbs/services/v2/checker/excel/ErrorAnalysisService.kt

```diff
@@ -13,29 +13,33 @@ import org.apache.poi.xssf.usermodel.XSSFCell
 import org.apache.poi.xssf.usermodel.XSSFWorkbook
 
 
-class PropagatedErrorsService(
+class ErrorAnalysisService(
     private val workbook: XSSFWorkbook,
     private val graph: ReferenceGraph, /* TODO get dependency graph from solution entry in db */
     private val solution: Map<Cell, String>, /* TODO maybe usa a kotlin set with indexing; get solution values from solution entry in db */
     private val handleService: HandlerService<ErrorAnalysisContext, Unit>? = null
 ) {
     private val evaluator: FormulaEvaluator = workbook.creationHelper.createFormulaEvaluator()
+    private val errors = HashSet<Cell>()
+    private val perrors = HashSet<Cell>()
+    private val visited = HashSet<Cell>()
 
-    fun findAllPropagatedErrors(outputCells: List<Cell>): Set<Cell> {
-        val errors = HashSet<Cell>()
-        val perrors = HashSet<Cell>()
-        val visited = HashSet<Cell>()
-        handleService?.getHandlers(When.BEFORE)?.forEach { handler -> handler.handle(ErrorAnalysisContext()) }
+
+    fun findAllErrors(outputCells: List<Cell>): Set<Cell> {
+        handleService?.getHandlers(When.BEFORE)
+            ?.forEach { handler -> handler.handle(ErrorAnalysisContext(errors, perrors)) }
         for (outputCell in outputCells) {
-            findPropagatedErrors(outputCell, errors, perrors, visited)
+            findErrors(outputCell)
         }
-        handleService?.getHandlers(When.AFTER)?.forEach { handler -> handler.handle(ErrorAnalysisContext()) }
+        handleService?.getHandlers(When.AFTER)
+            ?.forEach { handler -> handler.handle(ErrorAnalysisContext(errors, perrors)) }
```

Comment:
Instead of looping here, I would call a function on the `HandlerService' that runs all handlers of that type.